tests/test_audio_frequency_data.py

```diff
@@ -150,6 +150,20 @@ def test_magic_getitem_slice():
     npt.assert_allclose(FrequencyData(data[0], freqs)._data, freq[0]._data)
 
 
+def test_magic_getitem_error():
+    """
+    Test if indexing that would return a subset of the frequency bins raises a
+    key error.
+    """
+    freq = pf.FrequencyData([[0, 0, 0], [1, 1, 1]], [0, 1, 3])
+    # manually indexing too many dimensions
+    with pytest.raises(IndexError, match='Indexed dimensions must not exceed'):
+        freq[0, 1]
+    # indexing too many dimensions with ellipsis operator
+    with pytest.raises(IndexError, match='Indexed dimensions must not exceed'):
+        freq[0, 0, ..., 1]
+
```

Comment:
Additional unit test:
```suggestion

@pytest.mark.parametrize("shape", [(1,2), (1,2,3), (1,2,3,4)])
def test_magic_getitem_cannot_slice_freq(shape):
    frequencies = range(shape[-1])
    data = pf.FrequencyData(np.ones(shape), frequencies)

    last_cdim_slice = data[..., 0].freq
    equivalent_freq_slice = data.freq[...,0:1,:]
    last_datadim_slice = data.freq[..., 0] #= slice at specific frequency bin

    target_cdim_slice_shape = list(shape)
    target_cdim_slice_shape[-2] = 1

    assert(last_cdim_slice.shape != last_datadim_slice.shape)
    assert(last_cdim_slice.shape == equivalent_freq_slice.shape)
    assert(last_cdim_slice.shape == tuple(target_cdim_slice_shape))
```

Reply:
I think the test is wrong: ``0:1`` is not equal to ``0`` thats why its failing. if you fix line 171 its everthing is working as expected. If you slice with ``0`` the dimension is removed.
```suggestion
@pytest.mark.parametrize("shape", [(1,2), (1,2,3), (1,2,3,4)])
def test_magic_getitem_cannot_slice_freq(shape):
    frequencies = range(shape[-1])
    data = pf.FrequencyData(np.ones(shape), frequencies)
    last_cdim_slice = data[..., 0:1].freq
    equivalent_freq_slice = data.freq[...,0:1,:]
    last_datadim_slice = data.freq[..., 0] #= slice at specific frequency bin
    target_cdim_slice_shape = list(shape)
    target_cdim_slice_shape[-2] = 1
    assert(last_cdim_slice.shape != last_datadim_slice.shape)
    assert(last_cdim_slice.shape == equivalent_freq_slice.shape)
    assert(last_cdim_slice.shape == tuple(target_cdim_slice_shape))
```

Reply:
I just checked that as well. This also pertains to the `target_cdim_sice_shape`.  This would mean that the cshape can change due to slicing. If we consider this expected behaviour were fine. If not were in trouble...

Reply:
Sorry, my bad. I misinterpreted parts of the discussions last week.
Nevertheless, I think keeping the unit test is good, since it tests data with higher number of channel dimensions.

Reply:
I'm alread testing it here for 2D signals:
```python
def test_magic_getitem_ellipsis():
    """Test slicing operations by the magic function __getitem__."""
    signal = pf.Signal([[[1, 1, 1], [2, 2, 2]]], 44100)
    npt.assert_allclose(signal[..., 0].time, np.atleast_2d([1, 1, 1]))
    assert signal[..., 0].time.shape == (1, 3)
```

Reply:
We do have tests for 2D signals in there, maybe thats already enough.